internal/controller/helmrelease_controller.go

```diff
@@ -794,15 +794,22 @@ func (r *HelmReleaseReconciler) requestsForHelmChartChange(ctx context.Context,
 
 // event emits a Kubernetes event and forwards the event to notification controller if configured.
 func (r *HelmReleaseReconciler) event(_ context.Context, hr v2.HelmRelease, revision, severity, msg string) {
-	var meta map[string]string
-	if revision != "" {
-		meta = map[string]string{v2.GroupVersion.Group + "/revision": revision}
+	var eventMeta map[string]string
+
+	if revision != "" || hr.Status.LastAttemptedValuesChecksum != "" {
```

Comment:
The value is updated as soon as we have the checksum value, see: https://github.com/fluxcd/helm-controller/blob/main/internal/controller/helmrelease_controller.go#L324-L328

Reply:
Since `revision` and `token` are gonna be mutually exclusive I'm wondering if we should prefix their values with `revision/` and `token/` in the rate limiting key calculation in notification-controller

Reply:
Actually, maybe we should do this regardless of them being mutually exclusive 🤔

Reply:
You mean, prefix them _and_ always include them even if not found?

Reply:
I fail to see how this would matter when the SHA is just a calculation of the string join. Any change (or absence) in value (with or without prefix) would result in a new SHA.

Reply:
Yeah I'm thinking about the very unlikely scenario when the values of `revision` and `token` could be the same, from notification-controller point of view. The rate limiting key would be the same for both events, but that would be just a coincidence. The prefix would ensure that these events are considered different

Reply:
But this is unrelated to this PR anyways

Reply:
Ah, gotcha now.

It is important to note however that this would only have an effect if `revision=1` and then `token=1`, but revision becomes empty (or vice-versa). This would be highly coincidental, but may be worth a patch just for the sake of correctness. Should however not be blocking for the release we are planning.

Reply:
Created a PR for it, thanks @matheuscscp :bow: